### PR TITLE
Multiple cursor support with multi colored cursors

### PR DESCRIPTION
**Pull request by rnavagamuwa**

### CODE REVIEW

File: src/main/java/org/eclipse/che/ide/flux/liveedit/CheFluxLiveEditExtension.java [hunk 1]
```diff
@@ -65,9 +63,13 @@
...
 import org.eclipse.che.ide.resource.Path;
 
+import static org.eclipse.che.ide.api.notification.ReadState.READ;
+import static org.eclipse.che.ide.api.notification.StatusNotification.Status.SUCCESS;
+import static org.eclipse.che.ide.api.notification.StatusNotification.DisplayMode.FLOAT_MODE;
+
 
 @Extension(title = "Che Flux extension", version = "1.0.0")
-public class CheFluxLiveEditExtension {
+public class CheFluxLiveEditExtension implements CursorModelWithHandler, CursorActivityHandler {
```
Comment (sunix) [inline src/main/java/org/eclipse/che/ide/flux/liveedit/CheFluxLiveEditExtension.java]: Handler implementations should be in separate classes in my opinion, or use anonymous classes

File: src/main/java/org/eclipse/che/ide/flux/liveedit/CheFluxLiveEditExtension.java [hunk 2]
```diff
@@ -332,4 +395,54 @@ public void onDocumentChange(DocumentChangeEvent event) {
...
+            path = new Path(documentMain.getFile().getPath());
+            openedEditor = editorAgent.getOpenedEditor(path);
+            if (openedEditor instanceof TextEditorPresenter){
+                textEditor  = (TextEditorPresenter)openedEditor;
+            }
+            int offset = textEditor.getCursorOffset();
+            Log.info(CheFluxLiveEditExtension.class,offset);
+            /*here withUserName method sets the channel name and the withchannelName sets the username*/
+            /*if removed count equals to -100 that means there is only a cursor change */
+            Message liveResourceChangeMessage = new FluxMessageBuilder().with(documentMain).withOffset(offset).withRemovedCharCount(-100).withUserName(channelName).withChannelName(userId).buildLiveResourceChangeMessage();
```
Comment (sunix) [inline src/main/java/org/eclipse/che/ide/flux/liveedit/CheFluxLiveEditExtension.java]: You should create your own message type and not reusing "liveResourceChanged"
create another build method and use your own type: see https://github.com/sunix/che-plugin-flux-live-edit/blob/master/src/main/java/org/eclipse/che/ide/flux/liveedit/FluxMessageBuilder.java#L77

Comment (rnavagamuwa) [inline src/main/java/org/eclipse/che/ide/flux/liveedit/CheFluxLiveEditExtension.java]: sure will do that :+1:

File: che-plugin-flux-live-edit.iml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sunix) [inline che-plugin-flux-live-edit.iml]: You shouldn't add ide specific project files

Comment (rnavagamuwa) [inline che-plugin-flux-live-edit.iml]: I removed it

**Pull request closed without merging**